### ISSUE TITLE
A11Y: Improve labeling on the manage installed page

### ISSUE THREAD

**Issue by paul-geoghegan** [ubuntu/app-center#2163]
### Is there an existing issue for this?

- [x] I have searched the existing issues and found none matching what I'm reporting.

### Bug Description

Currently on the manage page the labeling could be improved for better a11y. Some of them are redundant, confusing, or just missing.

### Steps to reproduce the behavior

1. Tab through the manage page
2. Notice how it doesn't convay needed information to the user

### Expected behavior

Labels should explain things propperly for the user.
